db/view/view.cc

```diff
@@ -2515,12 +2515,17 @@ void view_builder::execute(build_step& step, exponential_backoff_retry r) {
     }).get();
 }
 
+future<> view_builder::mark_as_built(view_ptr view) {
+    return seastar::when_all_succeed(
+            _sys_ks.mark_view_as_built(view->ks_name(), view->cf_name()),
+            _sys_dist_ks.finish_view_build(view->ks_name(), view->cf_name())).discard_result();
```

Comment:
This changes the behavior subtly - before in just ONE of the duplicates, we first marked_view_as_built and only when that's done, we called finish_view_build. In the other place  when_all_succeed was already used. 
So I don't know if it's fine or not.

Reply:
It was first introduced by a73d9ccf31b59f06232c112204eb5c7a3c427cc2 and its "concurrent" version in view builder itself by 5f822e3928f9861b7040700e2214adca5744a3cd. I don't see why the former used the sequential approach other than Piotr didn't care to use when_all_succeed :shrug: